src/bundle/Resources/public/js/alloyeditor/src/toolbars/config/base.js

```diff
@@ -0,0 +1,115 @@
+import ReactDOM from 'react-dom';
+
+export default class EzConfigBase {
+    static outlineTotalWidth(block) {
+        let outlineOffset = parseInt(block.getComputedStyle('outline-offset'), 10),
+            outlineWidth = parseInt(block.getComputedStyle('outline-width'), 10);
+        if ( isNaN(outlineOffset) ) {
+            // Edge does not support offset-offset yet
+            // 1 comes from the stylesheet, see theme/alloyeditor/content.css
+            outlineOffset = 1;
+        }
+        return outlineOffset + outlineWidth;
+    }
+
+    static isEmpty(block) {
+        const count = block.$.childNodes.length;
+
+        return (count === 0 || (count === 1 && block.$.childNodes.item(0).localName === 'br'));
+    }
+
+    static setPositionFor(block, editor) {
+        const blockRect = block.getClientRect();
+        const outlineWidth = EzConfigBase.outlineTotalWidth(block);
+        const empty = EzConfigBase.isEmpty(block);
+        let positionReference = block;
+        let xy;
+        let domElement;
+        let range;
+        let left = 0;
+
+        if (editor.widgets.getByElement(block)) {
+            left = blockRect.left;
+        } else {
+            if (empty) {
+                block.appendHtml('<span>&nbsp;</span>');
+                positionReference = block.findOne('span');
+            }
+            range = document.createRange();
+            range.selectNodeContents(positionReference.$);
+            left = range.getBoundingClientRect().left;
+            if (empty) {
+                positionReference.remove();
+            }
+        }
+
+        xy = this.getWidgetXYPoint(
+            blockRect.left - outlineWidth,
+            blockRect.top + block.getWindow().getScrollPosition().y - outlineWidth,
+            CKEDITOR.SELECTION_BOTTOM_TO_TOP
+        );
+
+        domElement = new CKEDITOR.dom.element(ReactDOM.findDOMNode(this));
+        domElement.addClass('ae-toolbar-transition');
+        domElement.setStyles({
+            left: (left - outlineWidth) + 'px',
+            top: xy[1] + 'px'
+        });
+        return true;
```

Comment:
Empty line before

Reply:
I think we should try to incorporate the prettier - a tool for automatic fixing coding standards.